core/templates/services/prevent-page-unload-event.service.spec.ts

```diff
@@ -0,0 +1,96 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the preventPageUnloadEventService Service.
+ */
+
+import { TestBed } from '@angular/core/testing';
+import { PreventPageUnloadEventService } from 'services/prevent-page-unload-event.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+describe ('Prevent page unload event service', function() {
+  let preventPageUnloadEventService: PreventPageUnloadEventService = null;
+  let windowRef: WindowRef = null;
+
+  var reloadEvt = document.createEvent('Event');
+  reloadEvt.initEvent('mockbeforeunload', true, true);
+  reloadEvt.returnValue = null;
+  reloadEvt.preventDefault = () => {};
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [PreventPageUnloadEventService]
+    });
+    preventPageUnloadEventService = TestBed.get(PreventPageUnloadEventService);
+    windowRef = TestBed.get(WindowRef);
+  });
+
+  var mockWindow = {
+    addEventListener: function(eventname: string, callback: () => {}) {
+      document.addEventListener('mock' + eventname, callback);
+    },
+    location: {
+      reload: () => {
+        document.dispatchEvent(reloadEvt);
+      }
+    }
+  };
+
+  it('should adding listener', () => {
+    expect(preventPageUnloadEventService.getListenerActive()).toBe(false);
+
+    preventPageUnloadEventService.addListener();
+
+    expect(preventPageUnloadEventService.getListenerActive()).toBe(true);
+  });
+
+  it('should removing listener', () => {
+    expect(preventPageUnloadEventService.getListenerActive()).toBe(false);
+    preventPageUnloadEventService.addListener();
+    preventPageUnloadEventService.removeListener();
+
+    expect(preventPageUnloadEventService.getListenerActive()).toBe(false);
+  });
+
+  it('should test if Alert is displayed', () => {
+    spyOnProperty(windowRef, 'nativeWindow', 'get').and.returnValue(mockWindow);
+    preventPageUnloadEventService.addListener();
+    spyOn(reloadEvt, 'preventDefault');
+
+    mockWindow.location.reload();
+
+    expect(reloadEvt.preventDefault).toHaveBeenCalled();
+    expect(preventPageUnloadEventService.getListenerActive()).toBe(true);
+  });
+
+  it('should prevent multiple listeners', () => {
+    spyOn(window, 'addEventListener');
+
+    expect(window.addEventListener).toHaveBeenCalledTimes(0);
+    preventPageUnloadEventService.addListener();
+    expect(window.addEventListener).toHaveBeenCalledTimes(1);
+    expect(preventPageUnloadEventService.getListenerActive()).toBe(true);
+    preventPageUnloadEventService.addListener();
```

Comment:
Add a newline above to seperate the actions step from baseline verifications steps.

Reply:
DONE